pkg/asset/installconfig/aws/validation.go

```diff
@@ -0,0 +1,153 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/ipnet"
+	"github.com/openshift/installer/pkg/types"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+)
+
+// Validate executes platform-specific validation.
+func Validate(ctx context.Context, meta *Metadata, config *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	if config.Platform.AWS != nil {
+		allErrs = append(allErrs, validatePlatform(ctx, meta, field.NewPath("platform", "aws"), config.Platform.AWS, config.Networking, config.Publish)...)
+
+		if config.ControlPlane != nil && config.ControlPlane.Platform.AWS != nil {
+			allErrs = append(allErrs, validateMachinePool(ctx, meta, field.NewPath("controlPlane", "platform", "aws"), config.Platform.AWS, config.ControlPlane.Platform.AWS)...)
+		}
+		if len(config.Compute) > 0 {
```

Comment:
Do we need this guard?  I would expect the `range config.Compute` loop to be a no-op when the length was zero.

Reply:
yeah, will fix it.

Reply:
@wking fixed in https://github.com/openshift/installer/compare/a96182f3f4d7c7b69bd674bd918dd72a532f6ce4..9ce0bb39f5cbc2d581705794c8e16933cae350d4